lake table deserialize failed when fe restart

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. CREATE LAKE TABLE 
2. restart FE

### Expected behavior (Required)
success

### Real behavior (Required)
```
2022-07-04 21:50:28,154 ERROR (stateListener|79) [GlobalStateMgr.transferToMaster():902] failed to init journal after transfer to master! will exit
com.google.gson.JsonSyntaxException: com.google.protobuf.InvalidProtocolBufferException: While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.
        at com.google.gson.Gson.fromJson(Gson.java:978) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:928) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:877) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:848) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.catalog.lake.LakeTable.read(LakeTable.java:113) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Table.read(Table.java:259) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.CreateTableInfo.readFields(CreateTableInfo.java:72) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:193) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:143) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:186) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:1696) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.transferToMaster(GlobalStateMgr.java:893) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.access$800(GlobalStateMgr.java:261) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1619) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
Caused by: com.google.protobuf.InvalidProtocolBufferException: While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.
        at com.google.protobuf.InvalidProtocolBufferException.truncatedMessage(InvalidProtocolBufferException.java:107) ~[protobuf-java-3.16.1.jar:?]
        at com.google.protobuf.CodedInputStream$ArrayDecoder.pushLimit(CodedInputStream.java:1193) ~[protobuf-java-3.16.1.jar:?]
        at com.google.protobuf.CodedInputStream$ArrayDecoder.readMessage(CodedInputStream.java:887) ~[protobuf-java-3.16.1.jar:?]
        at com.staros.proto.ShardStorageInfo.<init>(ShardStorageInfo.java:56) ~[starproto-0.1.3.jar:?]
        at com.staros.proto.ShardStorageInfo.<init>(ShardStorageInfo.java:9) ~[starproto-0.1.3.jar:?]
        at com.staros.proto.ShardStorageInfo$1.parsePartialFrom(ShardStorageInfo.java:588) ~[starproto-0.1.3.jar:?]
        at com.staros.proto.ShardStorageInfo$1.parsePartialFrom(ShardStorageInfo.java:582) ~[starproto-0.1.3.jar:?]
        at com.google.protobuf.AbstractParser.parsePartialFrom(AbstractParser.java:158) ~[protobuf-java-3.16.1.jar:?]
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:191) ~[protobuf-java-3.16.1.jar:?]
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:203) ~[protobuf-java-3.16.1.jar:?]
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:208) ~[protobuf-java-3.16.1.jar:?]
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:48) ~[protobuf-java-3.16.1.jar:?]
        at com.staros.proto.ShardStorageInfo.parseFrom(ShardStorageInfo.java:216) ~[starproto-0.1.3.jar:?]
        at com.starrocks.catalog.lake.LakeTable.gsonPostProcess(LakeTable.java:100) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.gson.GsonUtils$ProcessHookTypeAdapterFactory$1.read(GsonUtils.java:441) ~[starrocks-fe.jar:?]
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/lake/LakeTable.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/lake/LakeTable.java
@@ -22,7 +22,6 @@ import java.io.DataOutput;
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.nio.charset.StandardCharsets;
 import java.util.List;
 
 /**
@@ -34,9 +33,9 @@ public class LakeTable extends OlapTable implements GsonPreProcessable {
     // Currently, storage group is table level, which stores all the tablets data and metadata of this table.
     // Format: service storage uri (from StarOS) + table id
     //
-    // "shardStorageInfoStr" is used for serialization of "shardStorageInfo".
-    @SerializedName(value = "shardStorageInfoStr")
-    private String shardStorageInfoStr;
+    // "shardStorageInfoBytes" is used for serialization of "shardStorageInfo".
+    @SerializedName(value = "shardStorageInfoBytes")
+    private byte[] shardStorageInfoBytes;
     private ShardStorageInfo shardStorageInfo;
 
     public LakeTable(long id, String tableName, List<Column> baseSchema, KeysType keysType, PartitionInfo partitionInfo,
@@ -83,14 +82,13 @@ public class LakeTable extends OlapTable implements GsonPreProcessable {
 
     @Override
     public void gsonPreProcess() throws IOException {
-        shardStorageInfoStr = new String(shardStorageInfo.toByteArray(), StandardCharsets.UTF_8);
+        shardStorageInfoBytes = shardStorageInfo.toByteArray();
     }
 
     @Override
     public void gsonPostProcess() throws IOException {
         super.gsonPostProcess();
-
-        shardStorageInfo = ShardStorageInfo.parseFrom(shardStorageInfoStr.getBytes(StandardCharsets.UTF_8));
+        shardStorageInfo = ShardStorageInfo.parseFrom(shardStorageInfoBytes);
     }
 
     @Override

--- a/fe/fe-core/src/test/java/com/starrocks/catalog/lake/LakeTableTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/catalog/lake/LakeTableTest.java
@@ -53,6 +53,7 @@ public class LakeTableTest {
         long tablet1Id = 10L;
         long tablet2Id = 11L;
         String serviceStorageUri = "s3://bucket/service/";
+        String endpoint = "region.host.com";
 
         // Schema
         List<Column> columns = Lists.newArrayList();
@@ -82,7 +83,8 @@ public class LakeTableTest {
         Deencapsulation.setField(table, "baseIndexId", indexId);
         table.addPartition(partition);
         table.setIndexMeta(indexId, "t1", columns, 0, 0, (short) 3, TStorageType.COLUMN, KeysType.AGG_KEYS);
-        ObjectStorageInfo objectStorageInfo = ObjectStorageInfo.newBuilder().setObjectUri(serviceStorageUri).build();
+        ObjectStorageInfo objectStorageInfo =
+                ObjectStorageInfo.newBuilder().setObjectUri(serviceStorageUri).setEndpoint(endpoint).build();
         ShardStorageInfo shardStorageInfo =
                 ShardStorageInfo.newBuilder().setObjectStorageInfo(objectStorageInfo).build();
         table.setShardStorageInfo(shardStorageInfo);
@@ -104,6 +106,8 @@ public class LakeTableTest {
         Assert.assertTrue(newTable.isLakeTable());
         LakeTable newLakeTable = (LakeTable) newTable;
         Assert.assertEquals(String.format("%s%d/", serviceStorageUri, tableId), newLakeTable.getStorageGroup());
+        ObjectStorageInfo newObjectStorageInfo = newLakeTable.getShardStorageInfo().getObjectStorageInfo();
+        Assert.assertEquals(endpoint, newObjectStorageInfo.getEndpoint());
 
         Partition p1 = newLakeTable.getPartition(partitionId);
         MaterializedIndex newIndex = p1.getBaseIndex();

